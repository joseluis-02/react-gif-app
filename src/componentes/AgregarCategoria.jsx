import { useState } from 'react';
import PropTypes from 'prop-types'
export const AgregarCategoria = ( { onNewCategory } ) => {

    const [InputValue, setInputValue] = useState('');
    const onInputChange = ( { target } ) => {
       // console.log(target.value);
        setInputValue( target.value );
    }

    const onSubmit = ( event ) => {
        event.preventDefault();
        //console.log( InputValue );
        if ( InputValue.trim().length <= 1 ) return;

        //setCategorias(c => [ InputValue,...c ]);
        onNewCategory( InputValue.trim() );

        //Limpiar el input
        setInputValue('');
    }

  return (
    <form onSubmit={ onSubmit } aria-label='form'>
        <input 
            type="text"
            placeholder="Buscar Gifs"
            value={ InputValue }
            onChange={ onInputChange }
        />
    </form>
    
  )
}


AgregarCategoria.propTypes = {
  onNewCategory: PropTypes.func.isRequired
}
