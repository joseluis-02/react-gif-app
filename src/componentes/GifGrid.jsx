import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types'

export const GifGrid = ( { Categoria } ) => {

    const {images, isLoading} = useFetchGifs( Categoria );

  return (
    <>
        <h3>{ Categoria }</h3>
        {
            isLoading && ( <h4>Cargando...</h4> )
        }
        <div className="card-grid">
            {
                images.map( ( image ) => {
                    return(
                        <GifItem
                            key={ image.id }
                            { ...image }
                        />
                    )
                })
            }
        </div>
    </>
  )
}

GifGrid.propTypes = {
    //Categoria: PropTypes.string.isRequireds,
    Categoria: PropTypes.string.isRequired

}
