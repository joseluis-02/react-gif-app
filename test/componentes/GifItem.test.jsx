import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/componentes/GifItem";

describe('Prueba de mi componente GifItem', () => {

    const title = 'One Punch';
    const url = 'https://midominio.com/imagen.jpg';

    test('Debe de hacer un snapshot del componente', () => { 

        const { container } = render(
            <GifItem 
                title={ title }
                url={ url }
            />
        );
        //console.log( container );

        expect(container ).toMatchSnapshot();

     });

     test('Debe mostrar la imagen con el url, alt correcto', () => { 

        render(
            <GifItem 
                title={ title }
                url= { url }
            />
        );

        const { src, alt} = screen.getByRole('img');
        //screen.debug();
        //console.log( src );

        expect( src ).toBe( url );
        expect( alt ).toBe( title );

      });

      test('Debe de mostrar el título en el componente en un párrafo', () => { 

        render(
            <GifItem 
                title={ title }
                url= { url }
            />
        );

        expect( screen.getAllByText( title ) ).toBeTruthy();

       });

});