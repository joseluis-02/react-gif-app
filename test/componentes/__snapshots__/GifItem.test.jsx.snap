// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de mi componente GifItem Debe de hacer un snapshot del componente 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Punch"
      src="https://midominio.com/imagen.jpg"
    />
    <p>
      One Punch
    </p>
  </div>
</div>
`;
